include/Game/Entities/Kabuto.h

```diff
@@ -2,9 +2,158 @@
+	virtual ~Obj() { }                                      // _1BC (weak)
+	virtual void setInitialSetting(EnemyInitialParamBase*); // _1C4
+	virtual void doUpdate();                                // _1CC
+	virtual void doDebugDraw(Graphics&);                    // _1EC
+	virtual void changeMaterial() = 0;                      // _200
+	virtual void initWalkSmokeEffect();                     // _230
+	virtual WalkSmokeEffect::Mgr* getWalkSmokeEffectMgr();  // _234
+	virtual EnemyTypeID::EEnemyTypeID getEnemyTypeID();     // _258 (weak)
+	virtual void doStartStoneState();                       // _2A4
+	virtual void doFinishStoneState();                      // _2A8
+	virtual void startCarcassMotion();                      // _2C4
+	virtual float getDownSmokeScale();                      // _2EC (weak)
+	virtual void doStartMovie();                            // _2F0
+	virtual void doEndMovie();                              // _2F4
+	virtual void setFSM(FSM*);                              // _2F8
+	virtual void createEffect();                            // _2FC (weak)
+	virtual void setupEffect();                             // _300 (weak)
+	virtual void startRotateEffect();                       // _304 (weak)
+	virtual void finishRotateEffect();                      // _308 (weak)
+	virtual void startWaitEffect();                         // _30C (weak)
+	virtual void finishWaitEffect();                        // _310 (weak)
+	virtual void effectDrawOn();                            // _314 (weak)
+	virtual void effectDrawOff();                           // _318 (weak)
+
+	void setRandTarget();
+	void getSearchedTarget();
+	void isAttackableTarget();
+	void createStoneAttack();
+	void updateCaution();
+	void getViewAngle();
+	void lifeIncrement();
+	void createRockEmitEffect();
+
+	// _00		= VTBL
+	// _00-_2BC = EnemyBase
+	FSM* m_kabutoFSM;                    // _2BC
+	WalkSmokeEffect::Mgr m_walkSmokeMgr; // _2C0
+	float _2C8;                          // _2C8
+	int _2CC;                            // _2CC
+	Vector3f m_targetPosition;           // _2D0
+	float _2DC;                          // _2DC
+	u8 _2E0;                             // _2E0, unknown
+	bool m_isUnderground;                // _2E1
+	u8 _2E2[0x2];                        // _2E2, unknown/probably padding
+	                                     // _2E4 = PelletView
+};
+
+struct Mgr : public EnemyMgrBase {
+	Mgr(int, u8);
+
+	virtual ~Mgr() { }                                 // _58 (weak)
+	virtual EnemyTypeID::EEnemyTypeID getEnemyTypeID() // _AC (weak)
+	{
+		return EnemyTypeID::EnemyID_Kabuto;
+	}
+	virtual SysShape::Model* createModel();         // _B0
+	virtual void loadModelData();                   // _C8
+	virtual void loadAnimData();                    // _CC
+	virtual J3DModelData* doLoadBmd(void* resource) // _D4 (weak)
+	{
+		return J3DModelLoaderDataBase::load(resource, 0x01240030);
+	}
+	virtual ResTIMG* getChangeTexture() = 0; // _E0
+
+	// _00		= VTBL
```

Comment:
Personally I opt for something like
// _00 - VTBL
// _04 -> 44 - EnemyMgrBase
But this is really minor, not serious at all